lib/puppet/functions/sqlserver/partial_params_args.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# @summary this function populates and returns the string of arguments which later gets injected in template.
+# arguments that return string holds is conditional and decided by the the input given to function.
+
+Puppet::Functions.create_function(:'sqlserver::partial_params_args') do
+  # @param args contains
+  #   Enum['ON', 'OFF'] $db_chaining
+  #   Enum['ON', 'OFF'] $trustworthy
+  #   String[1] $default_fulltext_language
+  #   String[1] $default_language
+  #   Optional[Enum['ON', 'OFF']] $nested_triggers
+  #   Optional[Enum['ON', 'OFF']] $transform_noise_words
+  #   Integer[1753, 9999] $two_digit_year_cutoff
+  #
+  # @return String
+  #   Generated on the basis of provided values.
+  #
```

Comment:
can we add some example how user passes inputs and what will be outcome.

Reply:
Sure. Thanks for pointing out. It will eventually increase the readability.